tests/console/yast2_samba.pm

```diff
@@ -14,258 +14,352 @@ use strict;
-
-    # change replication sleep and time-out
-    wait_screen_change { send_key 'alt-p' };
-    for (1 .. 10) { send_key 'down'; }
-    wait_screen_change { send_key 'alt-t' };
-    for (1 .. 2) { send_key 'up'; }
-
-    # change to not use SSL or TLS
-    wait_screen_change { send_key 'alt-u' };
-    for (1 .. 2) { send_key 'up'; }
-    assert_screen 'yast2_samba-server_ldap_advanced_expert_settings_not-use-ssl';
-    wait_screen_change { send_key 'ret' };
-    wait_screen_change { send_key 'alt-o' };
-
-    # now run test connection
-    send_key 'alt-t';
-    assert_screen 'yast2_samba-server_ldap_test-connection';
-    wait_screen_change { send_key 'alt-o' };
+    if ((is_sle(">=15")) || (is_leap(">=15.0")) || is_tumbleweed) {
+        record_soft_failure "bsc#1088152";
+        # _setup_ldap_in_samba; commented due to above bug
+    }
+    else {
+        _setup_ldap_in_samba;
+    }
 
     # finally, close with OK
     send_key 'alt-o';
-
     wait_serial('yast2-samba-server-status-0', 60) || die "'yast2 samba-server' didn't finish";
+
 }
 
-sub gui_new {
-    assert_screen 'yast2_samba-workgroup';
-    send_key 'alt-n';
-    assert_screen([qw(yast2_samba-firewall_active yast2_still_susefirewall2)]);
+sub setup_yast2_auth_server {
+    # workaround kernel message floating over console
+    assert_script_run "dmesg -n 4";
+
+    # check network at first
+    assert_script_run("if ! systemctl -q is-active network; then systemctl -q start network; fi");
+
+    script_run("yast2 auth-server; echo yast2-auth-server-status-\$? > /dev/$serialdev ", 0);
+
+    # check ldap server configuration started
+    assert_screen([qw(yast2_ldap_configuration_startup yast2_still_susefirewall2)], 60);
     if (match_has_tag 'yast2_still_susefirewall2') {
         record_soft_failure 'bsc#1064405';
         send_key 'alt-c';
-        assert_screen 'yast2_samba-firewall_active';
     }
-}
+    # only older version like SLES 12, Leap 42.3 as well as TW should still check the needle
+    assert_screen 'yast2_ldap_configuration_general-setting_firewall', 60;
+    send_key 'alt-e';
+
+    # configure stand-alone ldap server
+    assert_screen 'yast2_ldap_configuration_stand-alone';
+    send_key 'alt-n';
+
+    assert_screen 'yast2_ldap_configuration_stand-alone_tls';
+    # move to next page basic database settings and set base dn, ldap admin password
+    wait_screen_change { send_key 'alt-n' };
+    wait_screen_change { send_key 'alt-s' };
+    type_string 'dc=qa, dc=suse, dc=de';
+    wait_screen_change { send_key 'alt-a' };
+    for (1 .. 20) { send_key 'backspace'; }
+    type_string 'cn=admin';
+    wait_screen_change { send_key 'alt-l' };
+    type_string 'testing';
+    wait_screen_change { send_key 'alt-v' };
+    type_string 'testing';
+    # use database as default for ldap client
+    wait_screen_change { send_key 'alt-u' };
+    send_key 'alt-n';
+    assert_screen 'yast2_ldap_configuration_kerberos';
+    wait_screen_change { send_key 'alt-x' };
```

Comment:
I do not see the need for the `wait_screen_change` because immediately afterwards there is an assert_screen which should provide the synchronisation point